aion/web/elm/src/Ranking/View.elm

```diff
@@ -2,56 +2,116 @@ module Ranking.View exposing (..)
-                [ h4 [] [ text "General" ] ]
+            [ Grid.col [ Col.xs4, Col.offsetXs1 ]
+                [ Form.group []
+                    [ Form.label [] [ text "Select Category" ]
+                    , Select.select [ Select.onChange OnRankingCategoryChange ] (selectCategoriesAttributes model)
+                    ]
+                ]
             ]
         , Grid.row []
-            [ Grid.col [ Col.xs4 ]
+            [ Grid.col [ Col.xs10, Col.offsetXs1 ]
                 [ rankingTable model ]
             ]
         ]
 
 
+selectCategoriesAttributes : Model -> List (Select.Item msg)
+selectCategoriesAttributes model =
+    case model.rankingData.data of
+        RemoteData.Success rankingData ->
+            List.map (\category -> Select.item [ value (toString category.categoryId) ] [ text category.categoryName] ) rankingData.rankingList
+        _ -> []
+
+
 rankingTable : Model -> Html Msg
 rankingTable model =
-    Table.simpleTable
-        ( Table.simpleThead
-            [ Table.th [] [ text "User" ]
+    Table.table
+        { options = [ Table.striped, Table.hover ]
+        , thead = Table.simpleThead
+            [ Table.th [] [ text "#" ]
+            , Table.th [ Table.cellAttr(style [ ("width", "5%") ]) ] []
+            , Table.th [] [ text "User" ]
             , Table.th [] [ text "Score" ]
             ]
-        , displayScores model.rankingData
-        )
+        , tbody = displayScores model
+        }
 
 
-displayScores : WebData RankingData -> Table.TBody Msg
-displayScores rankingData =
-    case rankingData of
+displayScores : Model -> Table.TBody Msg
+displayScores model =
+    case model.rankingData.data of
         RemoteData.Success rankingData ->
             let
-                sortedScores =
-                    List.reverse (List.sortBy .score rankingData.data)
+                categoryScores = selectedCategoryScores rankingData.rankingList model.rankingData.selectedCategoryId
             in
-                Table.tbody [] (List.map displaySingleScore sortedScores)
+                Table.tbody [] (List.map (displaySingleScore model) (sortedScoresWithIndices categoryScores))
 
         _ ->
             Table.tbody [] []
 
 
-displaySingleScore : PlayerScore -> Table.Row msg
-displaySingleScore score =
-    Table.tr []
-        [ Table.td [] [ text score.userName ]
-        , Table.td [] [ text (toString score.score) ]
-        ]
+displaySingleScore : Model -> (Int, PlayerScore) -> Table.Row msg
+displaySingleScore model indexAndScore =
+    case indexAndScore of
+        (index, playerScore) ->
+            Table.tr []
+                [ Table.td [] [ text (toString (index + 1)) ]
+                , Table.td [] (scoreImage model.location index)
+                , Table.td [] [ text playerScore.userName ]
+                , Table.td [] [ text (toString playerScore.score) ]
+                ]
+
+
+scoreImage : Navigation.Location -> Int -> List (Html msg)
+scoreImage location index =
```

Comment:
1. You could export the medals to Constants or Urls (not sure we have a convention for that right now
2. You could then implement this function like this:
```
scoreImage : blabla
scoreImage location index = 
     case index of 
         0 -> [ img [src goldMedalPath] [] ]
         1 -> [...]
         2 -> [...]
         _ -> []
```
If you feel this gets repetitive you can always add
```
let 
    renderImage medal = [img [src medal] [] ]
in 
    case index of 
        0 -> renderImage goldMedalPath
        1 -> ...
        ...

Reply:
done